client/src/components/TaskList.jsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useMemo } from "react";
+import taskListCSS from "./TaskList.module.css";
+import api from "../clients/item-client.js";
+import Task from "./Task";
+import PropTypes from "prop-types";
+
+function TaskList({ tasks, setTasks, setIsLoading }) {
+   useEffect(() => {
+      (async () => {
+         setIsLoading(true);
+         try {
+            const allTasks = await api.getAllTasks();
+            allTasks ? setTasks(allTasks) : console.error("Server error");
+         } catch (error) {
+            console.error(error);
+         }
+         setIsLoading(false);
+      })();
+   }, [setTasks, setIsLoading]);
+
+   const tasksList = useMemo(
+      () =>
+         [...tasks].map((task) => (
```

Comment:
you don't really have to use spread operator here - it creates redundant extra array
```suggestion
         tasks.map((task) => (
```